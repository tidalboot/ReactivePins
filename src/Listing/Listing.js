import React, {Component} from 'react';
import './Listing.css';


class Listing extends Component {
  constructor(props) {
    super(props);
    this.state = {pin: 'Tracer', internationalFor: 'Listing for ', internationalFrom: ' from '}
    this.frenchify = this.frenchify.bind(this);
  };

  frenchify() {
    this.setState({ internationalFor: 'Inscription pour', internationalFrom: 'de ' })
  }

  render() {
   return (
   <div className="Listing">
     {this.state.internationalFor} {this.state.pin} {this.state.internationalFrom} {this.props.name}
     <br/>
     <button className="FrenchyButton" onClick={this.frenchify}>Frenchify!</button>
  </div>
   );
  }
}

export default Listing;