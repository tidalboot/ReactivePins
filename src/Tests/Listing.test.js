import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Listing from '../Listing/Listing';

describe('Listing tests', () => {
  const traderName = 'JeffKaplan';
  const mockFunction = jest.fn();
  const componentToTest = shallow(
    <Listing name={traderName} removeListing={mockFunction} />
  );

  it('shows JeffKaplan as the trader', () => {
    expect(componentToTest.text()).toContain(traderName);
  });

  it('shows that Tracer is the pin on offer', () => {
    expect(componentToTest.text()).toContain("Tracer");
  });

  it('renders and matches our existing snapshot', () => {
    const componentToTest = renderer.create(
      <Listing name={traderName} />
    );

    const componentAsJsonTree = componentToTest.toJSON();
    expect(componentAsJsonTree).toMatchSnapshot();
  });
});