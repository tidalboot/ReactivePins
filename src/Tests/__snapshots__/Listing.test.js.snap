// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing tests renders and matches our existing snapshot 1`] = `
<div
  className="Listing"
>
  Listing for 
   
  Tracer
   
   from 
   
  JeffKaplan
  <br />
  <button
    className="FrenchyButton"
    onClick={[Function]}
  >
    Frenchify!
  </button>
  <button
    className="RemoveListing"
    onClick={[Function]}
  >
    Remove listing
  </button>
</div>
`;
